src/heat_equations.jl

```diff
@@ -13,169 +19,517 @@ import ClimaCoupler:
+                    println("Analytical convergence factor: $analytic_conv_fac_value")
+                end
+            end
+            if plot_conv
+                gr()
+                plot()
+                color_dict, _ = get_color_dict()
+                color = color_dict[round(cs.model_sims.atmos_sim.params.a_i, digits=1)]
+                k_atm = 2:length(conv_fac_atm)+1
+                k_oce = 2:length(conv_fac_oce)+1
+                if atm && oce
+                    scatter!(k_atm, conv_fac_atm, label="atm", legend=legend, color=color, markershape=:x, markersize=5, xlabel="k", ylabel=ylabel, ylim=(0, maximum([maximum(conv_fac_atm[.!isnan.(conv_fac_atm)]), maximum(conv_fac_oce[.!isnan.(conv_fac_oce)])]) * 1.2))
+                    scatter!(k_oce, conv_fac_oce, label="oce", legend=legend, color=color, markershape=:circle, markersize=5)
+                elseif oce
+                    scatter!(k_oce, conv_fac_oce, label="oce", legend=legend, color=color, markershape=:circle, markersize=5, xlabel="k", ylabel=ylabel, ylim=(0, maximum(conv_fac_oce[.!isnan.(conv_fac_oce)]) * 1.2))
+                elseif atm
+                    scatter!(k_atm, conv_fac_atm, label="atm", legend=legend, color=color, markershape=:circle, markersize=5, xlabel="k", ylabel=ylabel, ylim=(0, maximum(conv_fac_atm[.!isnan.(conv_fac_atm)]) * 1.2))
+                end
+                if !isnothing(analytic_conv_fac_value)
+                    scatter!(k_atm, ones(length(k_atm)) * analytic_conv_fac_value, label="analytic", color=color, markershape=:hline, ylim=(0, analytic_conv_fac_value * 1.2))
+                end
+                display(current())
 
-function set_coupling_fields!(atmos_sim::HeatEquationAtmos, ocean_sim::HeatEquationOcean, atmos_T, ocean_T)
-    update_field!(atmos_sim, Val(:T_oce_sfc), ocean_T)
-    update_field!(ocean_sim, Val(:T_atm_sfc), atmos_T)
+            end
+            if return_conv
+                if atm && oce
+                    return conv_fac_atm, conv_fac_oce
+                elseif oce
+                    return nothing, conv_fac_oce
+                elseif atm
+                    return conv_fac_atm, nothing
+                end
+            end
+        end
+    end
 end
 
+# Update such that i can plot both atmosphere and ocean separately, via an oce boolean
+# argument. Fix the unstable plot, should be easier now when instability is defined for ocean and atmosphere separately.
+# Use is_stable in the unstable method. Then it should be done. Some additional comments and
+# Test with all combinations. Perhaps put things in new folders as well.
+function coupled_heat_equations(; iterate=10, parallel=false, boundary_mapping="mean", values=Dict{Symbol,Int}(), print_conv_facs_iter=false, plot_conv_facs_iter=false, analytic_conv_fac=false, atm=true, oce=true, combine=false, plot_unstable_range=false, a_is=[], var_name=nothing, xscale=:identity, legend=:right, log_conv_fac=false, xticks=nothing, text_scaling=(1, 5))
```

Comment:
- instead of `iterate`, `iterations` would be better.  `iterate` sounds like a boolean (True/False).
- `combine` is a very vague argument name considering what it actually does. maybe `combine_ρ_parallel` (also weird but better than the current name?)
- similar for `atm`, `oce` (sounds like "activate atmosphere/ocean component", not "Whether to consider the atmospheric convergence factor")
- would suggest `params` or similar instead of `values`, since you're expecting key-value pairs here.